internal/zinc-scripted/src/test/scala/sbt/internal/inc/IncHandler.scala

```diff
@@ -156,8 +156,8 @@ final class IncHandler(directory: File, cacheDir: File, scriptedLog: ManagedLogg
         val analysis = p.compile(i)
         p.discoverMainClasses(Some(analysis.apis)) match {
           case Seq(mainClassName) =>
-            val classpath = i.si.allJars :+ p.classesDir
-            val loader = ClasspathUtilities.makeLoader(classpath, i.si, directory)
+            val cp = p.classpath(i)
```

Comment:
This change is modifying the semantics of the classpath handling. You're including `internalClasspath` and `unmanagedJars` for something that before didn't require it.

Reply:
I advise to remove this change altogether. It's cluttering up the fix's diff for nothing.